cloudapi/v1/cloudapi.proto

```diff
@@ -6,49 +6,142 @@ import "google/protobuf/timestamp.proto";
 
 option go_package = "github.com/sourcegraph/cloud-api/go/cloudapi/v1;cloudapiv1";
 
-message ListInstancesRequest {}
+message Instance {
+  string id = 1;
+  optional InstanceDetails instance_details = 2;
+  optional PlatformDetails platform_details = 3;
+  optional InstanceState instance_state = 4;
+}
+
+enum InstanceType {
+  INSTANCE_TYPE_UNSPECIFIED = 0;
+  INSTANCE_TYPE_INTERNAL = 1;
+  INSTANCE_TYPE_TRIAL = 2;
+  INSTANCE_TYPE_PRODUCTION = 3;
+  INSTANCE_TYPE_EPHEMERAL = 4;
+}
+
+message InstanceDetails {
+  string name = 1;
+  InstanceType instance_type = 2;
+  string environment = 3;
```

Comment:
@michaellzc should we also make this enum? I expected fixed number of envs, and also this might help avoid using wrong types i.e. prod vs production 🤔

Reply:
let's keep it as string for now. otherwise adding new env will be coupling with API changes